### PR TITLE
CLI relock fix

### LINKED ISSUE #2455: Relock throws an error but succeeds
If you try to re-lock gold that is pending unlocking then it succeeds but shows an error:

```
james@accounts:~/census-ops$ celocli lockedgold:lock --from $CELO_VALIDATOR_ADDRESS --value 3000686416668556302541
Running Checks:
   ✔  Value [3000686416668556302541] is > 0
   ✔  687d149244d58fefdcc321b592b994472ba0080f is a registered Account
Running Checks:
   ✔  Account has at least 0 cGold
SendTransaction: relock
txHash: 0x788a73b4973e4177166d56122a34c9c4d46945b3e901a6a694887446841cfae2
Sending Transaction: relock... done
Sending Transaction: lock... !
Error: Returned error: gas required exceeds allowance or always failing transaction
    at Object.ErrorResponse (~/.nvm/versions/node/v10.18.0/lib/node_modules/@celo/celocli/node_modules/web3-core-helpers/src/errors.js:29:16)
    at ~/.nvm/versions/node/v10.18.0/lib/node_modules/@celo/celocli/node_modules/web3-core-requestmanager/src/index.js:140:36
    at callbackDecorator (~/.nvm/versions/node/v10.18.0/lib/node_modules/@celo/celocli/src/utils/eth-debug-provider.ts:16:7)
    at XMLHttpRequest.request.onreadystatechange (~/.nvm/versions/node/v10.18.0/lib/node_modules/@celo/celocli/node_modules/web3-providers-http/src/index.js:96:13)
    at XMLHttpRequestEventTarget.dispatchEvent (~/.nvm/versions/node/v10.18.0/lib/node_modules/@celo/celocli/node_modules/xhr2-cookies/xml-http-request-event-target.ts:44:13)
    at XMLHttpRequest._setReadyState (~/.nvm/versions/node/v10.18.0/lib/node_modules/@celo/celocli/node_modules/xhr2-cookies/xml-http-request.ts:219:8)
    at XMLHttpRequest._onHttpResponseEnd (~/.nvm/versions/node/v10.18.0/lib/node_modules/@celo/celocli/node_modules/xhr2-cookies/xml-http-request.ts:345:8)
    at IncomingMessage.<anonymous> (~/.nvm/versions/node/v10.18.0/lib/node_modules/@celo/celocli/node_modules/xhr2-cookies/xml-http-request.ts:311:39)
```

state before this command:
```
james@accounts:~/census-ops$ celocli lockedgold:show $CELO_VALIDATOR_ADDRESS
Running Checks:
   ✔  687d149244d58fefdcc321b592b994472ba0080f is a registered Account
lockedGold:
  total: 2170069882894147954 (~2.17 10^18)
  nonvoting: 0
  requirement: 0
pendingWithdrawals:
  0:
    time: 1579146743
    value: 3000686416668556302541 (~3000.69 10^18)
```

state after this command:
```
james@accounts:~/census-ops$ celocli lockedgold:show $CELO_VALIDATOR_ADDRESS
Running Checks:
   ✔  687d149244d58fefdcc321b592b994472ba0080f is a registered Account
lockedGold:
  total: 3002856486551450450495 (~3002.86 10^18)
  nonvoting: 3000686416668556302541 (~3000.69 10^18)
  requirement: 0
pendingWithdrawals:
```

### PR DESCRIPTION
**Pull request by mrsmkl**
### Description

Just added a check to make sure that locked value is not zero.

### Tested

Added unit test.

### Other changes

Trying to remove some flakiness from governance unit tests.

### Related issues

- Fixes #2455

### CODE REVIEW

File: packages/protocol/test/governance/governance.ts [hunk 1]
```diff
@@ -75,6 +75,15 @@ interface Transaction {
 // hard coded in ganache
 const EPOCH = 100
 
+async function mineToNextEpoch(web3) {
+  const bn = web3.eth.getBlockNumber()
+  if (bn % EPOCH < 50) {
+    await mineBlocks(EPOCH + 20, web3)
+  } else {
+    await mineBlocks(EPOCH - 20, web3)
+  }
```
Comment (yerdua) [inline packages/protocol/test/governance/governance.ts:84]: What's the rationale behind the +20 / -20?

File: packages/cli/src/commands/lockedgold/lock.ts [hunk 2]
```diff
@@ -47,7 +47,9 @@ export default class Lock extends BaseCommand {
     for (const txo of txos) {
       await displaySendTx('relock', txo, { from: address })
     }
-    const tx = lockedGold.lock()
-    await displaySendTx('lock', tx, { value: lockValue.toFixed() })
+    if (lockValue.gt(new BigNumber(0))) {
```
Comment (yerdua) [inline packages/cli/src/commands/lockedgold/lock.ts:50]: Shouldn't this already be covered by this check in line 33?

```.addCheck(`Value [${value.toFixed()}] is > 0`, () => value.gt(0))```

Comment (yerdua) [reply packages/cli/src/commands/lockedgold/lock.ts:50]: Nevermind. I see now that `lockValue` could be 0 if the value to be locked is covered by `relockValue`.

**Review by yerdua — APPROVED** [review]
I'm curious about a thing, but it's not blocking.

File: packages/protocol/test/governance/governance.ts (hunk 1 above)
Comment (mrsmkl) [reply packages/protocol/test/governance/governance.ts:84]: Well +20 is probably not useful, but -20 makes sure that the epoch won't change twice. Perhaps still need different handling for the case `bn % EPOCH == 0`.

**Pull request merged**